### ISSUE TITLE
Default button behavior incorrect on Mac desktop dialogs

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#6936]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.959
    OS Version      : macOS 10.15.4
    R Version       : 4.0.0

### Steps to reproduce the problem
- Problem happens with any Yes/No dialog in the code, using a couple of specific examples here
- Start Desktop IDE on Mac
- In Tools / Modify Keyboards Shortcuts dialog create a shortcut for the first entry (Browse Addins) so you can tell what's happening in next steps
- Click on "Reset..." button; this brings up a Yes/No dialog that was configured in the code to have "No" as the default button (i.e. the safest choice)
- Hit <kbd>enter</kbd>

### Describe the problem in detail
- Although keyboard focus goes to the No button by default, the Yes button is actually the default button (in macOS terms) as indicated by its blue fill; thus, hitting <kbd>Enter</kbd> triggers Yes and resets the shortcuts even though focus is on No; to trigger No you must use the spacebar

<img width="420" alt="screenshot of yes/no dialog showing focus on No and default on Yes" src="https://user-images.githubusercontent.com/10569626/82608420-839cfc00-9b6f-11ea-8a81-24ebd2239cfb.png">

### Describe the behavior you expected
- the code's request to make No the default should be honored both in terms of where focus goes and which button is marked as "default" in the OS; hitting enter or spacebar without any other action should do the No action in this case
- alternatively, might be better to not have a default button (in macOS terms) in these dialogs and have enter and spacebar both "click" the focused button

### Behavior when Yes is specified as default button
- If you do a long-running operation in the console, e.g. `Sys.sleep(100)` then Quit (Cmd+Q), you get dialog asking if you want to quit; the Yes button is specified as default in the code and is correctly set as the default at macOS level; however focus is still placed on the No button by default so spacebar would trigger that; ideally it would be placed on the Yes button

### Windows Desktop behavior
- Windows behaves correctly in terms of initial focus and by having enter and spacebar both activate the focused button

**Comment by gtritchie** [comment]
Don't think we need to do anything here. Might be nice to put focus on the default button to begin with but the behavior is consistent with overall macOS dialog behavior,.
